### DISCUSSION TITLE
Singular and plural mapping to the same term in glossary

### DISCUSSION THREAD

**Discussion by Zethson** [executablebooks/meta#738] [General] [answered]
Hey,

in many of my texts I am using the singular and plural of specific terms. For example `algorithm` and `algorithms`. I would like both of them to map to the same glossary term. Currently, I only have `algorithm` in my glossary and it looks ugly if I then only "terminize" algorithm part of algorithms and the s stays black/white and doesn't link to anywhere.

Is it possible to allow both, algorithm and algorithms to map to the same glossary element?

CC @annachristina

**Comment by choldgraf** [comment, marked as answer]
You should be able to provide multiple entries for a glossary definition. For example, see [the Jupyter Book glossary](https://jupyterbook.org/en/stable/reference/glossary.html). Entries for `MyST` and `MyST Markdown`. You can do this by putting the two terms on top of one another, e.g.:

```md
MyST
MyST Markdown
MySTs
    My definition.
```

One awkward thing about this is you'll have multiple terms listed even though some might just be the plural version of another.
Here are a few ways to get around the issue of having multiple terms just because you're using a plural:

**Use CSS to hide subsequent terms**.

The following example defines a CSS rule that will hide any term that comes immediately after a previous term.

````markdown
<style>

/* Do not display any glossary terms that are siblings of the first term */
dl.glossary dt + dt {
    display: none;
}

</style>


```{glossary}
Term
Terms
  Test text.
```
````
That should result in:

![image](https://user-images.githubusercontent.com/1839645/168257712-1e3aabbf-4b88-4d64-ba5e-c812fe81a37d.png)

**Structure your markdown to minimize the issue**:

1. Writing the `s` just outside of `{term}` to make it plural
2. Using Sphinx's cross-referencing syntax to choose your own text

An example of each:

```
{term}`Support Steward`s or {term}`Support Stewards <Support Steward>`.
```

becomes:

![image](https://user-images.githubusercontent.com/1839645/168255373-f85c1ebc-8b46-434b-b47a-eaa96daf3d14.png)

In my opinion the first is a little bit weird since the `s` is a different color, but folks will get it and it's not too disruptive, that's what I'd try.

**Reply by Zethson** [reply]
Ohh, great. Thank you very much!

**Reply by Zethson** [reply]
Is there a way to hide the definition in the Glossary though? 
![image](https://user-images.githubusercontent.com/21954664/167430973-898cd4e2-218f-4b21-bee2-deba1a631f86.png)

It doesn't look great in my opinion of both, the singular and plural version are rendered here.

**Reply by choldgraf** [reply]
Hmm - maybe you could write a little CSS rule that makes only the first entry in a group of glossary entries visible?

**Reply by choldgraf** [reply]
_moved above_

**Reply by Zethson** [reply]
Yeah, I used option 1 but wasn't happy about the coloring. Option 2 looks interesting though.

Thank you very much again for your suggestions! Much appreciated.

**Reply by choldgraf** [reply]
(moved above)

**Reply by Zethson** [reply]
Awesome, I love it!

**Reply by Zethson** [reply]
I have one issue with it. This would require all glossary terms to have two definitions. The single definitions get removed in any way. Is there a way to only apply this CSS rule if there are actual siblings?

I could add "dummy siblings", but it's not that pretty.

**Reply by Zethson** [reply]
Actually, another issue that this CSS code applies to all terms below the first one and it removes them completely. Only the most top term is actually being rendered.

**Reply by choldgraf** [reply]
Ah good catch - I've modified the CSS to use `+` instead of `~` and I think it should now work. I've also moved my answers up to the top in case others would benefit from them!

**Reply by Zethson** [reply]
Awesome, this works. Again, thank you!
